src/main/java/chess/domain/piece/Piece.java

```diff
@@ -0,0 +1,69 @@
+package chess.domain.piece;
+
+import chess.Strategy.MoveStrategy;
+import chess.domain.board.Board;
+import chess.domain.board.Position;
+
+import java.util.List;
+import java.util.Objects;
+
+public abstract class Piece {
+    private final String name;
+    private final Team team;
+    private final double score;
+    private final MoveStrategy moveStrategy;
+
+    public Piece(String name, Team team, double score, MoveStrategy moveStrategy) {
+        this.team = team;
+        this.name = convertName(name, team);
+        this.score = score;
+        this.moveStrategy = moveStrategy;
+    }
+
+    public abstract boolean isKing();
+
+    public abstract boolean isPawn();
```

Comment:
> 이 부분은 King, Pawn 제외 다른 Piece에도 모두 존재해야 하기에 수정하지 않았습니다!
> 
> `isKing()` 이 사용되는 부분은 `exitWhenPieceIsKing` 메서드의 다음과 같습니다.
> 
> `if (Objects.nonNull(destinationPiece) && destinationPiece.isKing()) {`
Piece를 목적지로 옮겼는데 해당 위치의 Piece가 King이면 게임이 끝나게하는 로직입니다.
해당 위치의 Piece가 Pawn, King이 아니라 다른 기물들일 수도 있기 때문에 모든 말에서 오버라이딩을 했습니다!

음... 네 다른 말들에도 있어야 하는데, isKing()은 King에서만 true고, isPawn()은 Pawn에서만 true면 다음과 같이 Piece에서 추상메서드가 아니라 기본 구현을 두면 되는거 아닌가 해서요.
```suggestion
    public boolean isKing() {
        return false; // king에서만 true로 오버라이딩
    }

    public boolean isPawn() {
        return false; // pawn에서만 true로 오버라이딩
    }
```

하위 구현체 별로 다른 로직이 아니라 중복으로 같은 로직(false return)을 선언해줘야해서 말씀드린거였어요! :)

Reply:
아!! 이해했습니다!!!!!